nxos_routing.tf

```diff
@@ -0,0 +1,214 @@
+              rule_name = route_map.name,
+              order     = entry.order,
+            }
+          ] : []
+        )
+      ]
+    ]
+  ])
+}
+
+resource "nxos_route_map_rule_entry_match_route" "route_map_rule_entry_match_route" {
+  for_each   = { for v in local.routing_route_map_rule_entry_match_route : v.key => v }
+  device     = each.value.device
+  rule_name  = each.value.rule_name
+  order      = each.value.order
+  depends_on = [nxos_route_map_rule_entry.route_map_rule_entry]
+}
+
+locals {
+  routing_route_map_rule_entry_match_route_prefix_list = flatten([
+    for device in local.devices : [
+      for route_map in try(local.device_config[device.name].routing.route_maps, []) : [
+        for entry in try(route_map.entries, []) : (
+          try(entry.match_prefix_list, null) != null ? [
+            {
+              key            = format("%s/%s/%s", device.name, route_map.name, entry.order),
+              device         = device.name,
+              rule_name      = route_map.name,
+              order          = entry.order,
+              prefix_list_dn = format("sys/rpm/pfxlistv4-[%s]", entry.match_prefix_list)
+            }
+          ] : []
+        )
+      ]
+    ]
+  ])
+}
+
+resource "nxos_route_map_rule_entry_match_route_prefix_list" "route_map_rule_entry_match_route_prefix_list" {
+  for_each       = { for v in local.routing_route_map_rule_entry_match_route_prefix_list : v.key => v }
+  device         = each.value.device
+  rule_name      = each.value.rule_name
+  order          = each.value.order
+  prefix_list_dn = each.value.prefix_list_dn
+  depends_on     = [nxos_route_map_rule_entry_match_route.route_map_rule_entry_match_route]
+}
+
+locals {
+  routing_route_map_rule_entry_set_regular_community = flatten([
+    for device in local.devices : [
+      for route_map in try(local.device_config[device.name].routing.route_maps, []) : [
+        for entry in try(route_map.entries, []) : [
+          for option in ["additive", "no_community", "set_criteria"] : (
+            try(entry[option], null) != null ? [
+              {
+                key          = format("%s/%s/%s/%s", device.name, route_map.name, entry.order, option),
+                device       = device.name,
+                rule_name    = route_map.name,
+                order        = entry.order,
+                additive     = try(entry.additive, "disabled"),
+                no_community = try(entry.no_community, "disabled"),
+                set_criteria = try(entry.set_criteria, "none")
+              }
+            ] : []
+          )
+        ]
+      ]
+    ]
+  ])
+}
+
+resource "nxos_route_map_rule_entry_set_regular_community" "route_map_rule_entry_set_regular_community" {
+  for_each     = { for v in local.routing_route_map_rule_entry_set_regular_community : v.key => v }
+  device       = each.value.device
+  rule_name    = each.value.rule_name
+  order        = each.value.order
+  additive     = each.value.additive
```

Comment:
Anything having "enabled"/"disabled", "yes"/"no" as value, should be normalized towards a boolean attribute in the (YAML) data model.